### ISSUE TITLE
GeoIP module is using a deprecated database

### ISSUE THREAD

**Issue by cfsmp3** [proftpd/proftpd#605]
MaxMind has been calling the GeoIP database ProFTPd uses "legacy" for a while now. I'm not sure when/if it will be discontinued, but if possibly ProFTPd's mod_geoip module should be updated so that users don't need to have the old database around.

Details on the change here:
https://dev.maxmind.com/geoip/geoip2/whats-new-in-geoip2/

**Comment by Castaglia** [comment]
See [Bug#4053](http://bugs.proftpd.org/show_bug.cgi?id=4053).   There's not been much perceived pressure to do this change yet, because of that open question of just when MaxMind will stop updating their legacy databases.

**Comment by pghmcfc** [comment]
The free legacy databases won't be updated after 1st April 2018, and the paid legacy databases on 2nd January 2019:

https://blog.maxmind.com/2018/01/02/discontinuation-of-the-geolite-legacy-databases/

**Comment by pghmcfc** [comment]
Ah, just looked at Bug#4053 and you knew that already; sorry!

**Comment by tspicer** [comment]
They officially removed the old database files. Our build failed today with a 404 error.

**Comment by Castaglia** [comment]
I've started working a replacement `mod_geoip2` module.  Please help me test it out; see https://github.com/Castaglia/proftpd-mod_geoip2.

**Comment by tspicer** [comment]
@Castaglia , do you suggest using libmaxminddb? https://github.com/maxmind/libmaxminddb and https://pkgs.alpinelinux.org/package/edge/main/x86/libmaxminddb

Been a while since I looked at this stuff, so I wanted to make sure I get the correct OS and other assets needed in place prior to testing.

**Comment by Castaglia** [comment]
Yes, the newer `mod_geoip2` uses libmaxminddb, and the free GeoIP2 databases from [here](https://dev.maxmind.com/geoip/geoip2/geolite2/).  It should work using Alpine, yes.

**Comment by Castaglia** [comment]
Question for the folks here:  Would you want/expect ProFTPD to support either `mod_geoip`, or `mod_geoip2`, _but not both at the same time_?  I'd like to get `mod_geoip2` into the 1.3.8 release, and am contemplating how to tweak the build system, to handle cases where users might want one or the other.

**Comment by pghmcfc** [comment]
Is there any point in supporting mod_geoip at all in 1.3.8? Is there any current support for the format it uses from any GeoIP database source?

**Comment by Castaglia** [comment]
@pghmcfc No current source that I'm aware of.  The only reason to keep `mod_geoip` around is in case there were sites which can't/won't update to using `libmaxminddb` and `mod_geoip2`.  Perhaps that's not really a concern, given how long it's been since the old `GeoIP` files were deprecated.

It's simpler for me to deprecate/remove `mod_geoip` in favor of `mod_geoip2`.  If you (and @hpreusse), as packagers, feel that that would easier for your perspective as well, then that's what we'll do.

**Comment by pghmcfc** [comment]
I may not be updating to `libmaxminddb` and `mod_geoip2` myself actually, since Maxmind's database license is no longer acceptable for Fedora. I'll probably just drop GeoIP support altogether unless a free database source appears that can be included in Fedora. I guess some people might want the support to be there to use with their own separately-sourced databases, but that would surely use `libmaxminddb` and `mod_geoip2`.

**Comment by hpreusse** [comment]
Please note that `geoip2` is not included in Debian yet, so there is no practical experience in using it; I've pushed it to the new queue. The license of libmaxminddb is "Apache License, Version 2.0", which ist listed on [DFSGLicenses](https://wiki.debian.org/DFSGLicenses), so I don't see an issue regarding the license.

As soon as `geoip2` is in Debian I can start deprecating the old `geoip` module. As I have no practical experience w/ `geoip` and `geoip2` I can't tell if there are technical reasons not to migrate.

**Comment by pghmcfc** [comment]
The license issue is with the Maxmind databases, not the library (https://bugzilla.redhat.com/show_bug.cgi?id=1786211). However, there may be an alternative database source: https://db-ip.com/db/lite.php, which has Creative Commons Attributions 4 International License and also supports MMDB format.

**Comment by hpreusse** [comment]
For accessing/downloading the DB we have a separate [package](https://packages.debian.org/geoipupdate). The alternative db sources do not seem to be packaged, although the license seems to be [DFSG compatible](https://wiki.debian.org/DFSGLicenses#Creative_Commons_Attribution_unported_.28CC-BY.29_v4.0).

**Comment by hpreusse** [comment]
Does it really make sense to package these data bases? They must be kept up to date to really work.

**Comment by pghmcfc** [comment]
They do need to be kept up to date, yes, but it's convenient for users that database updates come along the same path that regular package updates do.

**Comment by hpreusse** [comment]
So, this is rather a "packaging and how the distributions handle the issue" discussion. Regarding the answer to the question: I don't see any issue in removing `mod_geoip`, I expect `mod_geoip2` to be in Debian within 2-3 weeks.

**Comment by richarson** [comment]
Sorry for being so late to comment.

My use case is: we compile proftpd ourselves at $WORK and we're already using libmaxminddb for almost everything that we used to use the old geoip, except of course proftpd itself :)

I don't mind if you replace mod_geoip with mod_geoip2 or ship both and make them mutually exclusive, we won't be using both at the same time.

Thanks.
